nexus/src/nexus.rs

```diff
@@ -2249,6 +2259,49 @@ impl Nexus {
     pub async fn session_hard_delete(&self, token: String) -> DeleteResult {
         self.db_datastore.session_hard_delete(token).await
     }
+
+    pub async fn updates_refresh_metadata(&self) -> Result<(), Error> {
+        let rack = self.as_rack();
+        let trust_root = self
+            .tuf_trusted_root
+            .as_ref()
+            .ok_or_else(|| Error::InvalidRequest {
+                message: "updates system not configured".into(),
+            })?
+            .clone();
+
+        let artifacts = tokio::task::spawn_blocking(move || {
+            crate::updates::read_artifacts(&rack, &trust_root)
+        })
+        .await
+        // first, the JoinError:
+        .map_err(|e| Error::InternalError { internal_message: e.to_string() })?
+        // next, the boxed dyn Error:
+        .map_err(|e| Error::InternalError {
+            internal_message: e.to_string(),
```

Comment:
Nit: Could we add a little context here - just something like "we were trying to refresh updates, and saw {}"